ext/live_debugger.c

```diff
@@ -0,0 +1,1286 @@
+        ddtrace_create_capture_value(&EX(This), &capture_value, capture_config, capture_config->max_reference_depth);
+        ddog_snapshot_add_field(capture, DDOG_FIELD_TYPE_ARG, DDOG_CHARSLICE_C("this"), capture_value);
+    }
+}
+
+static void dd_log_probe_end(zend_ulong invocation, zend_execute_data *execute_data, zval *retval, void *auxiliary, void *dynamic) {
+    dd_log_probe_dyn *dyn = dynamic;
+    dd_log_probe_def *def = auxiliary;
+    UNUSED(invocation);
+
+    if (dyn->rejected) {
+        return;
+    }
+
+    if (def->parent.probe.evaluate_at == DDOG_EVALUATE_AT_EXIT && !dd_log_probe_eval_condition(def, execute_data, retval)) {
+        return;
+    }
+
+    ddog_Vec_SnapshotEvaluationError *errors;
+    zend_string *result = dd_eval_string(def->parent.probe.probe.log.segments, def->parent.probe.probe.log.capture, retval, &errors);
+    if (errors) {
+        dd_submit_probe_eval_error_snapshot(&def->parent.probe, errors);
+    }
+    ddog_CharSlice result_msg = dd_zend_string_to_CharSlice(result);
+    dd_log_probe_ensure_payload(dyn, def, &result_msg);
+
+    if (def->parent.probe.probe.log.capture_snapshot) {
+        DDTRACE_G(debugger_capture_arena) = dyn->capture_arena ? dyn->capture_arena : zend_arena_create(65536);
+        ddog_DebuggerCapture *capture = ddog_snapshot_exit(dyn->payload);
+        dd_log_probe_capture_snapshot(capture, def, execute_data);
+        const ddog_CaptureConfiguration *capture_config = def->parent.probe.probe.log.capture;
+        struct ddog_CaptureValue capture_value = {0};
+        ddtrace_create_capture_value(&EX(This), &capture_value, capture_config, capture_config->max_reference_depth);
+        ddog_snapshot_add_field(capture, DDOG_FIELD_TYPE_ARG, DDOG_CHARSLICE_C("@return"), capture_value);
+    }
+    ddtrace_sidecar_send_debugger_datum(dyn->payload);
+    if (DDTRACE_G(debugger_capture_arena)) {
+        zend_arena_destroy(DDTRACE_G(debugger_capture_arena));
+        DDTRACE_G(debugger_capture_arena) = NULL;
+    }
+    zend_string_release(result);
+    zend_string_release(dyn->service);
+}
+
+static bool dd_log_probe_begin(zend_ulong invocation, zend_execute_data *execute_data, void *auxiliary, void *dynamic) {
+    dd_log_probe_dyn *dyn = dynamic;
+    dd_log_probe_def *def = auxiliary;
+    UNUSED(invocation);
+
+    dd_probe_mark_active(&def->parent);
+
+    zval retval;
+    ZVAL_NULL(&retval);
+
+    dyn->payload = NULL;
+    dyn->rejected = def->parent.probe.evaluate_at == DDOG_EVALUATE_AT_ENTRY && !dd_log_probe_eval_condition(def, execute_data, &retval);
+    dyn->capture_arena = NULL;
+
+    if (!dyn->rejected && def->parent.probe.evaluate_at == DDOG_EVALUATE_AT_ENTRY) {
+        dd_log_probe_ensure_payload(dyn, def, NULL);
+        if (def->parent.probe.probe.log.capture_snapshot) {
+            ddog_DebuggerCapture *capture = ddog_snapshot_entry(dyn->payload);
+            DDTRACE_G(debugger_capture_arena) = zend_arena_create(65536);
```

Comment:
`DDTRACE_G(debugger_capture_arena)` can also be created in `ddtrace_collect_exception_debug_data`. Isn't there a risk of replacing the arena created there?

Reply:
This function is created called during serialization and dropped after serialization. No function calls should happen here.